lib/Backends/CPU/FunctionSpecializer.cpp

```diff
@@ -145,14 +160,16 @@ class FunctionSpecializer {
           specializedFTy->getReturnType() == FTy->getReturnType() &&
           "A function and its specialization should have the same return type");
       // The specialized function only takes non-constant parameters from the
-      // original function call. Check that the types of these parameter are the
-      // same for the original and the specialized function.
-      for (size_t argIdx = 0, e = F->arg_size(); argIdx < e; ++argIdx) {
-        if (!(argsToBeSpecialized & (((uint64_t)1) << argIdx))) {
-          assert(specializedFTy->getParamType(argIdx) ==
+      // original function call. Check that the types of these parameters are
+      // the same for the original and the specialized function.
+      for (size_t argIdx = 0, specializedArgIdx = 0, e = F->arg_size();
+           argIdx < e; ++argIdx) {
+        if (!isArgToBeSpecialized(argsToBeSpecialized, argIdx)) {
+          assert(specializedFTy->getParamType(specializedArgIdx) ==
```

Comment:
Just so I'm sure I understand what's going on with `specializedArgIdx` -- specialized args are removed from the arg list of the specialized function, so `specializedArgIdx` is only incremented after seeing unspecialized args? 

If so, would the naming here perhaps be more clear as `specializedFunArgIdx` or `unspecializedArgIdx`? Since these are indices for the unspecialized arguments I believe.

Reply:
Good catch! I improved the name of the variable and the comments. Please check if it is more understandable now.

Reply:
LGTM!